crates/apub/src/api/resolve_object.rs

```diff
@@ -67,14 +71,154 @@ async fn convert_response(
       }
       UserOrCommunity::Community(c) => {
         removed_or_deleted = c.deleted || c.removed;
-        res.community = Some(CommunityView::read(pool, c.id, local_user.as_ref(), false).await?)
+        res.community = Some(CommunityView::read(pool, c.id, local_user.as_ref(), is_admin).await?)
       }
     },
   };
-  // if the object was deleted from database, dont return it
-  if removed_or_deleted {
+
+  // if the object was deleted from database, don't return it for regular users
+  if removed_or_deleted && !is_admin {
```

Comment:
This if and variable removed_or_deleted actually seem unnecessary, because PostView, CommunityView etc will directly return NotFound if the object was deleted.

Reply:
`PostView` has an `is_mod_or_admin` argument to filter this, so it's redundant.
`CommentView` has no such argument, but it removes the comment contents. It would still return the comment without contents.
`PersonView` filters out all deleted users on SQL level, so it's redundant.
`CommunityView` has an `is_mod_or_admin` argument to filter this, so it's redundant.

The validation is probably better placed in the views, as that doesn't require validation in every place accessing the view, but for comments this is currently not checked there. Given that the content is removed, it may not be an issue to return a bit of metadata regardless, especially considering that it would be available through `/api/v3/comment` regardless.

I've removed the check from the resolve API now.